syntaxes/smithy.tmLanguage

```diff
@@ -674,20 +783,15 @@
             <key>name</key>
             <string>support.type.property-name.smithy</string>
             <key>match</key>
-            <string>[A-Z-a-z0-9_\.#$]+</string>
+            <string>[A-Z-a-z0-9_\.#$]+(?=\s*:)</string>
         </dict>
 
         <key>dquote_key</key>
         <dict>
             <key>name</key>
             <string>support.type.property-name.smithy</string>
-            <key>patterns</key>
-            <array>
-                <dict>
-                    <key>include</key>
-                    <string>#dquote</string>
```

Comment:
Didn't the dquote pattern handle escapes? Why not use this still?

Reply:
It doesn't really work (even on the main branch currently) because the styling from the import overrides the styling here, so there's no distinguishing between keys and values. If I were to copy the dquote rules over with just the styling change, it would take precedence over dquoted values, again leaving no distinction between keys and values. This is true even if I make it end with a colon, as since it's greedy and now spans lines it'll just keep consuming until it finds a dquote followed by a colon (which is very bad).

While we do lose styling on escapes, the whole string is still captured as the regex is greedy.

It's a trade-off. I'd rather have consistent styling between all keys that's different than the styling for values. And I might be able to make escapes work anyway.